base_classes/NXtransformations.nxdl.xml

```diff
@@ -176,6 +192,207 @@
 			differences.  Use of ``AXISNAME_end`` is recommended.
 		</doc>
 	</field>
+    <field name="distance" type="NX_FLOAT" units="NX_LENGTH" minOccurs="0">
```

Comment:
This fixes the coordinate frame's z axis direction. I think the frame should be left as a choice for beamlines/facilities.